pdr_backend/lake/persistent_data_store.py

```diff
@@ -31,6 +32,40 @@ def __init__(self, base_path: str, read_only: bool = False):
             database=f"{self.base_path}/duckdb.db", read_only=read_only
         )  # Keep a persistent connection
 
+        self.duckdb_data_types: Dict[DataTypeClass, str] = {
+            pl.Int8: "TINYINT",  # TINYINT in SQL typically represents an 8-bit integer.
+            pl.Int16: "SMALLINT",  # SMALLINT in SQL represents a 16-bit integer.
+            pl.Int32: "INTEGER",  # INTEGER is commonly used in SQL.
+            pl.Int64: "BIGINT",  # BIGINT in SQL represents a 64-bit integer.
+            pl.UInt8: "TINYINT",  # For unsigned, DuckDB may handle this similarly; check specifics.
+            pl.UInt16: "SMALLINT",  # Same note as above for unsigned small integer.
+            pl.UInt32: "INTEGER",  # Same note as above for unsigned integer.
+            pl.UInt64: "BIGINT",  # Same note as above for unsigned big integer.
+            pl.Float32: "FLOAT",  # FLOAT in SQL.
+            pl.Float64: "DOUBLE",  # DOUBLE in SQL for higher precision.
+            pl.Boolean: "BOOLEAN",  # BOOLEAN is standard in SQL.
+            pl.Utf8: "TEXT",  # TEXT for string data.
+            pl.Date: "DATE",  # DATE only for date; TIMESTAMP if including time.
+            pl.Datetime: "TIMESTAMP",  # TIMESTAMP for date and time.
+            pl.Time: "TIME",  # TIME only for time data.
+            pl.Duration: "INTERVAL",  # INTERVAL for time durations.
+            pl.List: "ARRAY",  # ARRAY for list data types, ensure compatibility.
+            pl.Object: "BLOB",  # BLOB for binary large objects, check usage context.
+            pl.Struct: "STRUCT",  # STRUCT for nested data, might need specific handling in SQL.
+            # Additional types like dictionaries, etc.,
+            # can be added based on specific needs and compatibility.
+        }
+
+    def _get_sql_column_type(self, column_type) -> str:
```

Comment:
why does this have to be created now, as part of this PR?

Reply:
This logic was created because we were using auto-casting for datatypes in the database. However, we now require at least one unique column, and making a column unique by altering it does not work. Therefore, we must specify it while creating the table. As a solution, I created this logic to cast column types manually.


I am open for different suggestions

Reply:
Can we just use the id column, which should be unique and every object has one at the moment?